html/code/modules/modules/xarinit.php

```diff
@@ -20,47 +20,41 @@
  */
 function modules_init()
 {
-    // Get database information
+    // Create tables inside a transaction
     $dbconn = xarDB::getConn();
-    $tables =& xarDB::getTables();
-
     $prefix = xarDB::getPrefix();
+    try {
+        $dbconn->begin();
+
+        sys::import('xaraya.tableddl');
+        xarXMLInstaller::createTable('table_schema-def', 'modules');
 
-    $tables['modules'] = $prefix . '_modules';
-    $tables['module_vars'] = $prefix . '_module_vars';
-    $tables['module_itemvars'] = $prefix . '_module_itemvars';
-    $tables['hooks'] = $prefix . '_hooks';
+        // We're done, commit
+        $dbconn->commit();
+    } catch (Exception $e) {
+        $dbconn->rollback();
+        throw $e;
+    }
+    // Get database information
+   // $dbconn = xarDB::getConn();
+    $tables =& xarDB::getTables();
     $tables['eventsystem'] = $prefix . '_eventsystem';
```

Comment:
Do we need this line?

Reply:
We need this to overcome following error:
File     : html/lib/xaraya/events.php
Line     : 509
Undefined index: eventsystem